src/explore-education-statistics-frontend/src/modules/api-subscriptions/NewSubscriptionPage.tsx

```diff
@@ -0,0 +1,139 @@
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import Button from '@common/components/Button';
+import FormFieldTextInput from '@common/components/form/FormFieldTextInput';
+import Form from '@common/components/form/Form';
+import FormProvider from '@common/components/form/FormProvider';
+import Panel from '@common/components/Panel';
+import { mapFieldErrors } from '@common/validation/serverValidations';
+import Yup from '@common/validation/yup';
+import { Dictionary } from '@common/types';
+import Page from '@frontend/components/Page';
+import withAxiosHandler from '@frontend/middleware/ssr/withAxiosHandler';
+import dataSetFileQueries from '@frontend/queries/dataSetFileQueries';
+import { DataSetFile } from '@frontend/services/dataSetFileService';
+import apiNotificationService from '@frontend/services/apiNotificationService';
+import React from 'react';
+import { GetServerSideProps, NextPage } from 'next';
+import Head from 'next/head';
+import { QueryClient } from '@tanstack/react-query';
+
+interface FormValues {
+  dataSetId: string;
+  email: string;
+}
+
+const errorMappings = [
+  mapFieldErrors<FormValues>({
+    target: 'dataSetId',
+    messages: {
+      ApiPendingSubscriptionAlreadyExists:
+        'The user is already subscribed to the API data set.',
+      ApiVerifiedSubscriptionAlreadyExists:
+        'The user already has a pending subscription for the API data set. They must verify their subscription.',
+    },
+  }),
+];
+
+interface Props {
+  dataSetFile: DataSetFile;
+}
+
+const NewSubscriptionPage: NextPage<Props> = ({ dataSetFile }) => {
+  const { id, title } = dataSetFile;
+
+  const handleFormSubmit = async ({ email }: FormValues) => {
+    await apiNotificationService.requestPendingSubscription({
+      email,
+      dataSetId: id,
```

Comment:
Is this the correct `id` here? I'm wondering if it should be `dataSetFile.api.id`?

It might be worth renaming `id` to whatever type of id it is.

Same comment applies to the de-referencing of `id` from `dataSetFile` in the other pages.

Reply:
I think it is meant to be `dataSetId.id` as that's the id we need to fetch the data set details on the confirmation page, if I try with `dataSetFile.api.id` the request works but the link to the confirmation page in the email has the api id in which can't then be used to fetch the details.

Reply:
I've just checked with the backend code and you are correct, thanks. There are that many different types of id's relating to files and data sets that it's getting very confusing! 😆 

On the backend the Notifier knows nothing about the Public API data model, so although the parameter `dataSetId` sounds like it would be a Public API data set id, like you say, it needs to be the same type of id that's used to link back to the data catalogue page. On the backend this is a `File.DataSetFileId`! 😆